services/spar/src/Spar/Scim/User.hs

```diff
@@ -406,6 +403,23 @@ calculateVersion uidText usr = Scim.Weak (Text.pack (show h))
     h :: Digest SHA256
     h = hashlazy (Aeson.encode (Scim.WithId uidText usr))
 
+{-|
+Check that the UserRef is not taken; or that it's taken by the given user id.
+
+FUTUREWORK: This check is very slow as it scans the every user for the team and decodes JSON
+to get the externalId. This could be cleaned up eventually by adding an index.
+-}
+assertUserRefUnused :: UserId -> SAML.UserRef -> Scim.ScimHandler Spar ()
+assertUserRefUnused wireUserId userRef = do
+  mExistingUserId <- lift $ wrapMonadClient (Data.getUser userRef)
+  case mExistingUserId of
+    -- No existing user for this userRef; it's okay to set it
+    Nothing -> return () 
+    -- A user exists; verify that it's the same user before updating
+    Just existingUserId  ->
+      unless (existingUserId == wireUserId) $
```

Comment:
You can avoid the case and test for `mExistingUserId == Just wireUserId` here instead.

Reply:
Are you sure? I want to throw the error if and only if this is a `Just` AND the IDs are NOT equal;

I can't do `existingUserId /= Just wireUserId` because then it would throw if a `Nothing` is returned (which should be fine.

Reply:
you're right, sorry, my mistake.  no easy way to simplify this!